src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MinLengthAttribute.cs

```diff
@@ -56,24 +56,19 @@ public override bool IsValid(object value)
             {
                 return true;
             }
-            var str = value as string;
-            if (str != null)
+            if (value is string str)
             {
                 length = str.Length;
             }
             else
             {
-                ICollection collection = value as ICollection;
-
-                if (collection != null)
+                if (value is ICollection collection)
                 {
                     length = collection.Count;
                 }
                 else
                 {
-                    // A cast exception previously occurred if a non-{string|array} property was passed
-                    // in so preserve this behavior if the value does not implement ICollection
-                    length = ((Array)value).Length;
+                    throw new InvalidCastException(SR.Format(SR.LengthAttribute_InvalidValueType, value.GetType()));
```

Comment:
>throw new InvalidCastException(SR.Format(SR.LengthAttribute_InvalidValueType, value.GetType())); [](start = 20, length = 96)

As for MaxLengthAttribute: what if it is an array? You're removing functionality here which was added to solve a previous issue.